capstone/fitness/src/main/java/com/google/sps/servlets/UpdateProgressServlet.java

```diff
@@ -0,0 +1,144 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;  
+import com.google.sps.servlets.*;
+import com.google.sps.util.*;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/update-progress")
+public class UpdateProgressServlet extends HttpServlet {
+
+  static Gson gson;  
+  
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    // Display the complete session screen which then sends a POST to this server.
+    RequestDispatcher view = request.getRequestDispatcher("complete-session.html");
+    view.forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    
+    // Get the users email.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    // Redirect if user not found.
+    if(userEmail == null) {
+      RequestDispatcher view = request.getRequestDispatcher("/");
+      view.forward(request, response);
+      System.out.println("user is null");
```

Comment:
better use a logging library instead if you need to log for debugging purposes and also include more details, eg. user email. Also, if forward() happens, will this line still execute?